codex-rs/tui/src/app.rs

```diff
@@ -413,6 +429,172 @@ impl App {
         self.config.model_reasoning_effort = effort;
     }
 
+    fn handle_apply_approval_preset(&mut self, preset: ApprovalPreset) -> Result<()> {
+        let approval = preset.approval;
+        let sandbox = preset.sandbox;
+
+        match sandbox.clone() {
```

Comment:
Please move all this code to another file: this is too much to add to `app.rs`.

Also, there should be no need to `clone()` this.